### PR TITLE
Update to more recent cardano-ledger-specs.

### PR DESCRIPTION
**Pull request by nc6**
This re-enables the PBFT VRF checks. These now seem to be passing
consensus tests.

Note that currently this references an unmerged branch - I want to verify the changes here before re-enabling this back in the ledger specs.

### CODE REVIEW

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Protocol.hs [hunk 1]
```diff
@@ -447,7 +448,8 @@ mkShelleyGlobals epochInfo TPraosParams {..} = SL.Globals {
 -- | Check whether this node meets the leader threshold to issue a block.
 meetsLeaderThreshold
   :: forall c.
-     ConsensusConfig (TPraos c)
+     (VRF.VRFAlgorithm (VRF c))
```
Comment (mrBliss) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Protocol.hs:451]: We use either `Crypto c` or `TPraosCrypto c` throughout `ouroboros-consensus-shelley`. I know you only need VRF here, but for consistency I would go for one of those other constraints.

**Comment by mrBliss** [comment]
It seems @dcoutts got to it first (in our absence): https://github.com/input-output-hk/ouroboros-network/pull/2324

**Comment by mrBliss** [comment]
Oops, I hadn't realised https://github.com/input-output-hk/cardano-ledger-specs/pull/1590 hadn't been merged yet. Reopening, but this PR will need updating.

**Comment by nc6** [comment]
OK, this has been rebased and repointed to the latest c-l-s. VRF checks are enabled, and we pick up some bug fixes along the way :-)

**Comment by mrBliss** [comment]
CI fails with: `VRFKeyBadNonce` in the ThreadNet tests.

**Comment by nc6** [comment]
Urgh, I ran the tests before and thought they worked...

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Protocol/State.hs [hunk 2]
```diff
@@ -154,7 +167,7 @@ rewind toSlot st
         Nothing      -> older
         Just current -> Map.insert toSlot current older
 
-empty :: WithOrigin SlotNo -> STS.State (STS.PRTCL c) -> TPraosState c
+empty :: WithOrigin SlotNo -> (SL.ChainDepState c) -> TPraosState c
```
Comment (mrBliss) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Protocol/State.hs:170]:
```suggestion
empty :: WithOrigin SlotNo -> SL.ChainDepState c -> TPraosState c
```

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Protocol/State.hs [hunk 3]
```diff
@@ -97,13 +98,25 @@ lastSlot st
 -- calling this to have a state containing more history than needed.
 append
   :: SlotNo
-  -> STS.State (STS.PRTCL c)
+  -> (SL.ChainDepState c)
```
Comment (mrBliss) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Protocol/State.hs:101]:
```suggestion
  -> SL.ChainDepState c
```

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs [hunk 4]
```diff
@@ -202,12 +204,18 @@ protocolInfoShelley genesis initialNonce maxMajorPV protVer mbCredentials =
...
     initChainDepState :: State.TPraosState c
     initChainDepState = State.empty Origin $
-      SL.PrtclState
-         (SL.chainOCertIssue     initShelleyState)
-         (SL.chainEpochNonce     initShelleyState)
-         (SL.chainEvolvingNonce  initShelleyState)
-         (SL.chainCandidateNonce initShelleyState)
-         (SL.chainPrevEpochNonce initShelleyState)
+      SL.ChainDepState
+        {
```
Comment (mrBliss) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs:208]:
```suggestion
      SL.ChainDepState {
```

**Comment by nc6** [comment]
bors merge

**Pull request merged**